app/Http/Controllers/EmpresaController.php

```diff
@@ -56,19 +56,22 @@ public function register(Request $request)
             'nome' => 'required|unique',
             'nome_social' => 'required|string|max:50',
             'razao_social' => 'required|string|max:50',
-            'cnpj' => 'required|integer|max:12', 
+            'cnpj' => 'required|integer|max:12|unique', 
             'telefone' => 'required|integer|max:11',
             'email'    => 'required|string|email'
         ]);
 
-        $user_id         = $request->input('user_id');
-        $nome            = $request->input('nome');
-        $nome_social     = $request->input('nome_social');
-        $razao_social    = $request->input('razao_social');
-        $cnpj            = $request->input('cnpj');
-        $telefone        = $request->input('telefone');
-        $email           = $request->input('email');
-
+        $user_id              = $request->input('user_id');
+        $nome                 = $request->input('nome');
+        $nome_social          = $request->input('nome_social');
+        $razao_social         = $request->input('razao_social');
+        $cnpj                 = $request->input('cnpj');
+        $telefone             = $request->input('telefone');
+        $email                = $request->input('email');
+        $tipo_empresa_id      = $request->input('tipo_empresa_id');
+        $natureza_empresa_id  = $request->input('natureza_empresa_id');
+        $inscricao_empresa_id = $request->input('inscricao_empresa_id');
+        
          try{
```

Comment:
Não precisa colocar dois try e catch dentro do mesmo metodo

Reply:
você fala do controller e services?

se for eu tirei do services e deixei so no controller.